### DISCUSSION TITLE
Dependencies for containers

### DISCUSSION THREAD

**Discussion by margual56** [emilk/egui#2330] [Q&A] [answered]
Hi! I am trying to:
- Use a github action on my project
- Use [cross-rs](https://github.com/cross-rs/cross) to cross-compile my project (uses docker)

Both of those fail because the dependencies listed in the readme are not enough for a docker container. Which other dependencies do I need? It keeps telling me that it needs `gdk`, `fontconfig` and others...

**Comment by kyzyl** [comment, marked as answer]
Cross has a pre-build hook to run stuff before cross attempts to build. Adding the following to your `Cargo.toml` or `Cross.toml` should fix the fontconfig dependency:

```toml
[package.metadata.cross.target.x86_64-unknown-linux-gnu]
pre-build = [
    "dpkg --add-architecture $CROSS_DEB_ARCH",
    "apt-get update && apt-get --assume-yes install libfontconfig1-dev"
]
```
Of course the exact commands and packages will depend on the target. The above works for ubuntu which you are likely running for CI.

Alternatively you could pre-bake a custom docker image with whatever you need, push it to ghcr.io and add `image = "ghcr.io/<your org>/<target_name>-cross:main"` under the appropriate cross target config. Then cross will attempt to pull down the image and use it instead of the default image for that target.

**Reply by margual56** [reply]
Thanks! It works now, that was the missing dependency, at least in `ubuntu-latest`. This is the full list of dependencies I have:
`libfontconfig1-dev libxcb-render0-dev libxcb-shape0-dev libxcb-xfixes0-dev libspeechd-dev libxkbcommon-dev libssl-dev librust-atk-dev libwebkit2gtk-4.0-dev libsoup2.4-dev`.

I'm sure there are some of them that are redundant, but hey... It works! :)
